buildSrc/src/main/java/com/osacky/flank/gradle/YamlConfigWriterTask.kt

```diff
@@ -2,20 +2,21 @@ package com.osacky.flank.gradle
 
 import org.gradle.api.DefaultTask
 import org.gradle.api.file.ProjectLayout
+import org.gradle.api.file.RegularFile
+import org.gradle.api.model.ObjectFactory
+import org.gradle.api.provider.Property
 import org.gradle.api.tasks.Internal
 import org.gradle.api.tasks.OutputFile
 import org.gradle.api.tasks.TaskAction
-import java.io.File
 import javax.inject.Inject
 
-open class YamlConfigWriterTask @Inject constructor(private val config: FladleConfig, private val extension: FlankGradleExtension, projectLayout: ProjectLayout) : DefaultTask() {
+open class YamlConfigWriterTask @Inject constructor(private val config: FladleConfig, private val extension: FlankGradleExtension, projectLayout: ProjectLayout, objectFactory: ObjectFactory) : DefaultTask() {
 
   private val yamlWriter = YamlWriter()
 
-  private val fladleDir = projectLayout.fladleDir.get().asFile
-
   @OutputFile
-  val fladleConfigFile: File = fladleDir.resolve("flank.yml")
+  val flankConfig: Property<RegularFile> =
+    objectFactory.fileProperty().convention(projectLayout.fladleDir.map { it.file("flank.yml") })
```

Comment:
what does `projectLayout.fladleDir.map { it.file("flank.yml") }` return? can we just use that?

Reply:
That would be a Provider, not a Property

Reply:
Ah, I just took a look at this and I think I understand it better now. Is it possible to only pass the string of the config name as an input to this task and then that can be used to compute the the output directory?
Otherwise we are only using that name as an output. That is, two tasks with the same inputs will have different output directories.

Reply:
What's the issue with that?
What we need is to declare the inputs. Tight now it's not possible because the extension is not serializable but also, I don't think the extension is the right data to provide to the yaml writer.

Let me move config to an input, so at least we'll fix what you are asking for.